pallets/proposals/src/tests.rs

```diff
@@ -431,13 +431,108 @@ fn withdraw_removes_project_after_all_funds_taken() {
             RuntimeOrigin::signed(*ALICE),
             project_key
         ));
+
+        if let Some((account, projects)) = CompletedProjects::<Test>::iter().next() {
+             assert_eq!(projects.len(),1);
+            assert_eq!(projects.contains(&project_key),true);
+        }
+
         assert!(
             Projects::<Test>::get(project_key).is_none(),
             "Project should have been removed after funds withdrawn."
         )
     });
 }
 
+#[test]
+fn store_project_info_after_project_is_completed() {
+    build_test_externality().execute_with(|| {
+        let cont = get_contributions::<Test>(vec![*BOB], 100_000);
+        let prop_milestones = get_milestones(1);
+        let project_key = create_project::<Test>(*ALICE, cont, prop_milestones, CurrencyId::Native);
+        let milestone_key = 0;
+        let _ =
+            Proposals::submit_milestone(RuntimeOrigin::signed(*ALICE), project_key, milestone_key)
+                .unwrap();
+        let _ = Proposals::vote_on_milestone(
+            RuntimeOrigin::signed(*BOB),
+            project_key,
+            milestone_key,
+            true,
+        )
+            .unwrap();
+        assert!(Projects::<Test>::get(project_key).is_some());
+        assert_ok!(Proposals::withdraw(
+            RuntimeOrigin::signed(*ALICE),
+            project_key
+        ));
+
+        if let Some((_account, projects)) = CompletedProjects::<Test>::iter().next() {
+            assert_eq!(projects.len(),1);
+            assert_eq!(projects.contains(&project_key),true);
+        }
+    });
+}
+
+#[test]
+fn store_too_many_projects_for_account() {
+    build_test_externality().execute_with(|| {
+        let max = <Test as Config>::MaxProjectsPerAccount::get();
+        let cont = get_contributions::<Test>(vec![*BOB], 100_000);
+        let prop_milestones = get_milestones(1);
+        let milestone_key = 0;
+        (0..=max).for_each(|i| {
+            if i != max {
+                let project_key = create_project::<Test>(
```

Comment:
This can be simplified, you are repeating the code to create and vote and withdraw in both arms. 
If you take it out the if and else statements and just place it in the loop it will be cleaner. 

Then you can just do the assertions in the arms.

Reply:
Done